includes/options/class-amp-options-manager.php

```diff
@@ -90,11 +91,46 @@ public static function register_settings() {
 								'type'     => 'string',
 								'readonly' => true,
 							],
+							Option::READER_THEME         => [
+								'type' => 'string',
+							],
 						],
 					],
 				],
 			]
 		);
+
+		add_filter( 'rest_pre_get_setting', [ __CLASS__, 'rest_get_options' ], 10, 2 );
+	}
+
+	/**
+	 * In REST, the option will be invalid and returned as null if its schema doesn't exactly match what's defined in
```

Comment:
Can you give an example of an option returning null? This function will return all the options, even if it is not declared in the REST schema.

Reply:
This kept happening as I was switching between this branch, which adds the READER_THEME option, and an older branch that didn't have the option. On the older branch, the setting kept coming back over REST as null because the settings object in my local DB had the READER_THEME field set. This appears to be intentional core behavior implemented here: 

https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php#L336

If you follow the stack trace there, you end up here: https://github.com/WordPress/WordPress/blob/d1015f46b5f578f047a6f1ecc22a41ceb0ef1436/wp-includes/rest-api.php#L1343 Which returns an error (resulting in `null` in the REST response) if `additionalProperties` is set at all. 

I didn't know about this before working on this issue, but it was interesting to learn that the REST settings controller expects a setting's schema to match exactly what's in the database.

Reply:
Hmm, interesting indeed. Thanks for the explanation!

Reply:
This is probably to prevent unwanted values from being injected into the REST fields via the DB.

Reply:
As discussed over chat, this issue resulted in `amp-options` being `null` in the response for me.

Something to consider is adding a new AMP endpoint specific for its options, rather than re-using core's options endpoint. Then we'll have more flexibility possibly. Similar to how @pierlon created a new rest field for storing the `amp_enabled` state for a given post, rather than rely on the core-provided postmeta field.

Reply:
This may be especially relevant as the plugin suppression logic currently stores the data in the `amp-options` setting under `suppressed_plugins` with a mapping of plugin slugs to data about the suppressed plugin state. However, the REST API endpoint wouldn't be exposing that data for mutation. So we'd need some endpoint to just manage the array of plugin slugs are suppressed, and then another endpoint to get suppression details. (At least as I see it right now.)